databroker/assets/mongo.py

```diff
@@ -104,10 +102,9 @@ def _datum_col(self):
 
     @property
     def _connection(self):
-        if self.__conn is None:
-            self.__conn = MongoClient(self.config['host'],
-                                      self.config.get('port', None))
-        return self.__conn
+        if self.__db is None:
+            self.__db = _get_mongo_database(self.config)
+        return self.__db.client
```

Comment:
Can we use `self._db.client` and skip the `None` check above?

Reply:
I don't understand the reason for this change, can you explain?

Reply:
If we access `self._db`, it perhaps the `if self.__db is None: ...` check  itself, and we can reduce the repeated code.

Reply:
updated